docs/contributing/release_guide.rst

```diff
@@ -5,170 +5,349 @@ Release Guide
+   doing exactly one task!
+
+Several weeks before release
+----------------------------
+
+* Create an issue on GitHub for the release with a checklist of tasks
+  to be performed.
+
+.. Should we create an issue template for this?  Should we convert the
+   checklist into an issue format?
+
+* Three weeks before the release, announce that a feature freeze will
+  occur one week before the anticipated release date. Only pull requests
+  with a limited scope that do not significantly change functionality
+  should be merged during the feature freeze. |exclude bugfix|
+
+* Announce a code freeze beginning ∼2–3 weekdays before the release.
+  Only bug fixes and pull requests directly related to the release
+  should be merged during the code freeze. |exclude bugfix|
 
 Pre-release
 -----------
 
-* Enact a feature freeze that begins approximately one week before the
-  anticipated release date.  Only pull requests with a limited scope that
-  do not significantly change functionality should be merged during this
-  time period.
+* Create a pull request to revise changelog entries to make sure that
+  they are understandable and correctly categorized.
+
+  .. tip::
+
+     Apply the :guilabel:`No changelog entry needed` label to pull
+     requests that change multiple changelog entries in order to skip
+     the changelog entry check.
+
+* Open a pull request to update hooks in :file:`.pre-commit-config.yaml`
+  to the most recent versions, and run ``pre-commit run --all-files`` to
+  apply all changes. |exclude bugfix|
+
+* Open a pull request to re-run pre-executed notebooks, including those
+  for charged particle radiography.
+
+* Begin an upload to Zenodo_ for the new release using the
+  ``team@plasmapy.org`` login, and reserve a digital object identifier
+  (DOI).
+
+* Open a pull request to update :file:`docs/about/citation.rst` to
+  reflect the new version, and include the reserved DOI.
+
+.. Should we switch to Citation File Format?  It appears to be better
+   supported by Zenodo.
+
+* Open a pull request to update and alphabetize the author list in
+  :file:`docs/about/credits.rst`. Add ORCID_ identifiers, if needed.
+
+* Open a pull request to update :file:`codemeta.json`. Update the author
+  list, version, and other metadata, as needed. Update the
+  ``"identifier"`` tag with the DOI for the new release.
+
+* Open a pull request to update :file:`.mailmap`. (Add bash command for this?)
+
+* Run ``make linkcheck`` in :file:`docs/`, and if necessary, open a pull
+  request to update redirects and fix broken links. The reserved DOI
+  link in :file:`docs/about/citation.rst` will not work yet since the
+  Zenodo_ record will not be published until after the official release.
+
+  .. tip::
+
+     Use ``linkcheck_allowed_redirects`` in :file:`docs/conf.py` to
+     specify allowed redirects. For example, DOI links are more
+     persistent than most hyperlinks, but are always redirects.
+
+* Open a pull request to update the changelog.
```

Comment:
@rocco8773's suggestion is to perform the towncrier magic on the release branch and then to merge it back in to main.

For the release, we have to delete `dev` from toctree for the changelog index page, but the `dev` should be added back in the PR back to `main`.

(though can we do an `if` thing for this so we don't have to do this step?)

Reply:
See https://github.com/BaPSF/bapsflib/pull/92 and https://github.com/BaPSF/bapsflib/issue/80